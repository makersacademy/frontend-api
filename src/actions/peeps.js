import store from '../store/configureStore'

export const addPeep = (peep) => ({
  type: 'ADD_PEEP',
  peep
})

export const startAddPeep = (peepData = {}) => {
  return (dispatch) => {
    const session = store.getState().sessions
    const data = new FormData()
    const requestHeader = {
      method: "POST",
      headers: new Headers({
        'Authorization': `Token token=${session.session_key}`,
        'Content-Type': 'application/json'
      }),
      body: data.append("json", JSON.stringify({"peep": {"user_id": session.user_id, ...peepData}))
    }
    return fetch("https://chitter-backend-api.herokuapp.com/peeps", requestHeader).then(response => {
      return response.json()
    }).then(json => {
      dispatch(addPeep(json))
    }).catch(e => console.log(e))
  }
}

export const removePeep = ({id} = {}) => ({
  type: 'REMOVE_PEEP',
  id
})

export const setPeeps = (peeps) => ({
  type: 'SET_PEEPS',
  peeps
})

export const startSetPeeps = () => {
  return (dispatch) => {
    return fetch("https://chitter-backend-api.herokuapp.com/peeps").then(response => {
      return response.json()
    }).then(json => {
      dispatch(setPeeps(json))
    }).catch((e) => console.log(e))
  }
}
