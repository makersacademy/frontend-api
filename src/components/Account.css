.ui.card {
  margin: auto !important;
  width: 100% !important;
  height: 100% !important;
}

div.signout.extra.content a {
  color: #4183c4 !important;
}