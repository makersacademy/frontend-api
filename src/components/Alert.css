.ui.message {
  width: 100% !important;
  position: absolute !important;
  display: none;
  bottom: 0;
  left: 0;
  opacity: 0;
  transition: all 2s ease-in-out !important;
}

