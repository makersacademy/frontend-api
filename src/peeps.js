$(document).ready(function(){
  $.get('https://chitter-backend-api.herokuapp.com/peeps', function(data) {
    data.forEach(function(element) {
      $(".peeps").append("<li>" + element.body + "</li>");
    });

    
  });

  // $(document).ready(function(){
  //   $("#btn1").click(function(){
  //     $("p").append(" <b>Appended text</b>.");
  //   });
  //
  //   $("#btn2").click(function(){
  //     $("ol").append("<li>Appended item</li>");
  //   });
  // });
  //
});
