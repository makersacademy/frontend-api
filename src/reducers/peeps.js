const peepsReducerDefaultState = [];

export default (state=peepsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_PEEP':
      return [action.peep, ...state]
    case 'REMOVE_PEEP':
      return state.filter(({id}) => id !== action.id)
    // case 'EDIT_PEEP':
    //   return state.map((peep) => peep.id === action.id ? {...peep, ...action.updates} ? peep;
    case 'SET_PEEPS':
      return action.peeps
    case 'LIKE_PEEP':
      // const index = state.findIndex((peep) => peep.id === action.id))
      state[state.findIndex((peep) => peep.id === action.id)].likes.push(action.user)
      return state
    default:
      return state
  }
}
