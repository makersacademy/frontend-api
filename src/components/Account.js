import React from "react";
import './Account.css'

const Account = ({ session, setSession }) => {

  const signOut = () => {
    localStorage.removeItem('user')
    setSession(false)
  }

  return (
    <div className="profile">
      <div className="ui card">
        <div className="image">
          <img src={session.avatar} alt='my beautiful pic'/>
        </div>
        <div className="content">
          <a href="#" className="header">{session.handle}</a>
          <div className="meta">
            <span className="date">Joined in 2019</span>
          </div>
          <div className="description">
            <div>Art Director - Los Angeles</div>
            <div>UCLA '15</div>
          </div>
        </div>
        <div className="extra content">
          <a>
            <i className="user icon"></i>
            22 Friends
          </a>
        </div>
        <div className="signout extra content">
          <a className="blue" href="#" onClick={signOut}>Sign Out</a>
        </div>
    </div>
  </div>
  )
}

export default Account