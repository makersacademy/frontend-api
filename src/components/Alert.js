import React, { useEffect } from "react";
import './Alert.css'

const Alert = ({ alert, setAlert }) => {
  
  useEffect(() => {

    if (alert) {
      const flash = document.querySelector('.flashAlert')
      if (flash) {
        flash.style.display = 'block'
        flash.style.opacity = 1;
      }
  
      let fadeTimeout = setTimeout(() => {
        if (flash) {
          flash.style.opacity = 0;
        }
      }, 2000)
  
      let showTimeout = setTimeout(() => {
        if (flash) {
          flash.style.display = 'none';
        }
      }, 4000)
  
      return () => {
        setAlert(false)
        const flash = document.querySelector('.flashAlert')
        clearTimeout(fadeTimeout)
        clearTimeout(showTimeout)
        if (flash) {
          flash.style.display = 'none'
        }
      }
    }

  }, [alert, setAlert])

    return (
      <div className={`ui ${alert.type} message flashAlert`} >
        <div className="header">
          {alert.message}
        </div>
      </div>
    )
 
}

export default Alert