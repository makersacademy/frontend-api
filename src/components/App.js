import React, { useState, useEffect } from "react";
import Menu from './Menu'
import LogIn from  './LogIn'
import Feed from './Feed'
import Peep from './Peep'
import Alert from './Alert' 

const App = () => {

  const [option, setOption] = useState('Feed');
  const [session, setSession] = useState(false)
  const [alert, setAlert] = useState(false)
  const [redirect, setRedirect] = useState(false)


  useEffect(() => {
    const user = localStorage.getItem('user')
    if (user) {
      setSession(JSON.parse(user))
    }
  }, [])

  return (
    <div className="app"> 
      <div className="ui top-bar">
        <Menu onSetOption={setOption} session={session} redirect={redirect} setRedirect={setRedirect} />
      </div>
      <div className='ui container content-field'>
        {
        option === 'LogIn' ? <LogIn setSession={setSession} session={session} setAlert={setAlert} />  :
        option === 'Feed' ? <Feed /> :
        option === 'Peep' ? <Peep redirect={setRedirect}/>
        : ''
        }
      <Alert alert={alert} setAlert={setAlert} />  
      </div>
    </div>
  )
}

export default App

