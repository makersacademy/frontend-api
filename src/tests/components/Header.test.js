import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header'
import peeps from '../fixtures/peeps'

let wrapper

beforeEach(() => {
  wrapper = shallow(<Header />)
})

test('should render Feed correctly', () => {
  expect(wrapper).toMatchSnapshot();
})
