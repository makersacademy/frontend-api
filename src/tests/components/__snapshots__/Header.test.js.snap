// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Feed correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <a
    className="navbar-brand"
    href="#"
  >
    Chitter
  </a>
  <button
    aria-controls="navbarNav"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarNav"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarNav"
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="#"
        >
          Home 
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
        >
          Features
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
        >
          Pricing
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link disabled"
          href="#"
        >
          Disabled
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
